### PR TITLE
Playground susch / different features

### PR DESCRIPTION
**Pull request by susch19**
* added enumeration modifiable list to reduce the amount of locks required
* added wauzi spawn egg function
* added interaction with wauzi
* made preview of item textures always the same file
* added multiple camera angles
* added zoom hot key
* added changeable fov

### CODE REVIEW

File: OctoAwesome/OctoAwesome.Basics/FunctionBlocks/Chest.cs [hunk 1]
```diff
@@ -61,6 +66,8 @@ private void UiComponentChanged((ComponentContainer, string, bool show) e)
             AnimationComponent.AnimationSpeed = -60f;
             changedSub?.Dispose();
 
+            changedSub?.Dispose();
+
```
Comment (jvbsl) [inline OctoAwesome/OctoAwesome.Basics/FunctionBlocks/Chest.cs]:
```suggestion
```

File: OctoAwesome/OctoAwesome.Client/Components/IViewCreator.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+/// <summary>
+/// Interface for creating a camera view.
+/// </summary>
+public interface IViewCreator
+{
+    /// <summary>
+    /// Gets a value indicating whether this creator creates a first person camera view.
+    /// </summary>
+    public bool IsFirstPerson { get; }
```
Comment (jvbsl) [inline OctoAwesome/OctoAwesome.Client/Components/IViewCreator.cs]:
```suggestion
    bool IsFirstPerson { get; }
```

File: OctoAwesome/OctoAwesome.Client/Components/IViewCreator.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    /// <summary>
+    /// Creates camera view for a entity with a head.
+    /// </summary>
+    /// <param name="head">The head component of the entity.</param>
+    /// <param name="playerPos">The position of the entity.</param>
+    /// <returns>Tuple of camera position, camera look at point and camera up vector.</returns>
+    public (Vector3 cameraPosition, Vector3 lookAt, Vector3 cameraUpVector) CreateCameraUpVectorAndView(HeadComponent head, Coordinate playerPos);
```
Comment (jvbsl) [inline OctoAwesome/OctoAwesome.Client/Components/IViewCreator.cs]:
```suggestion
    (Vector3 cameraPosition, Vector3 lookAt, Vector3 cameraUpVector) CreateCameraUpVectorAndView(HeadComponent head, Coordinate playerPos);
```

File: OctoAwesome/OctoAwesome.Basics/Definitions/Items/WauziItem.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            simulationRelay.OnNext(new EntityNotification(EntityNotification.ActionType.Add, wauzi));
+
+            return 0;
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            simulationSource.Dispose();
+            simulationRelay?.Dispose();
```
Comment (jvbsl) [inline OctoAwesome/OctoAwesome.Basics/Definitions/Items/WauziItem.cs]:
```suggestion
            simulationRelay.Dispose();
```

File: OctoAwesome/OctoAwesome.Basics/Entities/WauziEntity.cs [hunk 5]
```diff
@@ -16,45 +18,130 @@ namespace OctoAwesome.Basics.Entities
...
+                }
+                var shortestDist = new Index2(wauziPosition.Position.GlobalBlockIndex).ShortestDistanceXY(lastGoal, size);
+                move = new Vector2(shortestDist.X, shortestDist.Y);
+                return move;
+            }
+        }
+
         /// <summary>
-        /// Gets or sets a value indicating the time left to the next jump.
+        /// Gets or sets the time left for jumping.
```
Comment (jvbsl) [inline OctoAwesome/OctoAwesome.Basics/Entities/WauziEntity.cs]:
```suggestion
        /// Gets or sets a value indicating the time left to the next jump.
```

File: OctoAwesome/OctoAwesome.Basics/Entities/WauziEntity.cs [hunk 6]
```diff
@@ -16,45 +18,130 @@ namespace OctoAwesome.Basics.Entities
...
                 return;
 
-            controller.JumpInput = true;
+            Controller.JumpInput = true;
             if (!Components.Contains<RelatedEntityComponent>())
             {
                 var relEntity = new RelatedEntityComponent();
                 relEntity.RelatedEntityId = entity.Id;
-                Components.AddIfTypeNotExists(relEntity);
+                Components.Add(relEntity);
```
Comment (jvbsl) [inline OctoAwesome/OctoAwesome.Basics/Entities/WauziEntity.cs:144]: Is that change correct?

**Review by jvbsl — CHANGES_REQUESTED** [review]

File: OctoAwesome/OctoAwesome.Basics/Entities/WauziEntity.cs (hunk 6 above)
Comment (susch19) [reply OctoAwesome/OctoAwesome.Basics/Entities/WauziEntity.cs:144]: I have no clue, have to check

Comment (susch19) [reply OctoAwesome/OctoAwesome.Basics/Entities/WauziEntity.cs:144]: Has already been checked in line 140

**Review by jvbsl — APPROVED** [review]

**Pull request merged**